### PR TITLE
WIP chore(build): add experiment version bumping to release scripts

### PR DESCRIPTION
**Pull request by vladikoff**

### CODE REVIEW

File: grunttasks/experiments.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function (grunt) {
+  var PACKAGE_JSON = '../package.json';
+
+  /**
+   *
+   */
+  grunt.registerTask('experiments', 'Updates experiment configuration for train deploys', function () {
+    var fs = require('fs');
```
Comment (pdehaan) [inline grunttasks/experiments.js]:
```
grunttasks/experiments.js
  12:8  error  fs is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

File: grunttasks/experiments.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    var semverUtils = require('semver-utils');
+    var lib = require('./lib/experiments');
+
+    var version = require(PACKAGE_JSON).version;
```
Comment (pdehaan) [inline grunttasks/experiments.js]: I think we're already injecting package.json in the [Gruntfile.js](https://github.com/mozilla/fxa-content-server/blob/c2f05420f6444a4e5ad3c9a0aca2c0b0c5cdb518/Gruntfile.js#L31-L33):

```js
  grunt.initConfig({
    pkg: grunt.file.readJSON('package.json')
  });
```

Not sure if you can nuke line 6, 12, 16, and then change line 17 below to somethin' like:
```js
var semver = semverUtils.parse('<%= pkg.version %>');
```

File: grunttasks/lib/experiments.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var EXPERIMENTS_FILE = './server/config/production-experiments.json';
+
+module.exports = {
+  generateNewConfig: function(branchName) {
+    var trainExperiments = 'github:mozilla/fxa-content-experiments#' + branchName;
+
+    var experimentsFile = JSON.parse(fs.readFileSync(EXPERIMENTS_FILE));
```
Comment (pdehaan) [inline grunttasks/lib/experiments.js]:
```js
var experimentsFile = grunt.file.readJSON(EXPERIMENTS_FILE);
```

File: package.json [hunk 4]
```diff
@@ -87,6 +88,7 @@
     "grunt-todo": "0.4.0",
     "husky": "0.8.1",
     "intern": "git://github.com/theintern/intern#42aebd9beb942a11e2c6e6d7687c70a1c22b9bf7",
+    "semver-utils": "^1.1.0",
```
Comment (pdehaan) [inline package.json]: Drop the `^`

File: grunttasks/lib/experiments.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var EXPERIMENTS_FILE = './server/config/production-experiments.json';
```
Comment (pdehaan) [inline grunttasks/lib/experiments.js]: This `EXPERIMENTS_FILE` value is duplicated here and in the [./grunttasks/bump.js](https://github.com/mozilla/fxa-content-server/pull/2725/files#diff-becc724e0df9a64e398a42d62bc54ffdR14) file.
Not sure if we can normalize in one place and pass as a param or something.

**Comment by vladikoff** [comment]
Thanks for taking a look at this @pdehaan, mostly pushed what I had so far, still a bit of TODO :)

**Comment by jrgm** [comment]
@vladikoff so you mentioned the issue of setting `experiments.dir` (/home/app/experiments in prod) and I had previously said that I thought [convict](https://github.com/mozilla/node-convict) did not do deeper merging. But it appears I was wrong and if `a.json` and `b.json` both define an `experiments` hash then you get a merge of the two (with `b.json` trumping for any values it declares). @zaach is that right? 

But anyways, it seems fine to just have `production-experiments.json` set dir to /home/app/experiments and remove the `experiments` section form the puppet-config.

**Comment by vladikoff** [comment]
Blocked by https://github.com/mozilla/fxa-content-server/pull/2725

**Comment by vladikoff** [comment]
Blocked by https://github.com/mozilla/fxa-content-server/pull/2727

**Comment by vladikoff** [comment]
Moving this to Train 43, going to create the branch manually for now

**Comment by shane-tomlinson** [comment]
@vladikoff - #2727 has merged, does this have any other dependencies? Do you still want to try to get this in this train?

**Comment by vladikoff** [comment]
> @vladikoff - #2727 has merged, does this have any other dependencies? Do you still want to try to get this in this train?

Yes, targeting train 43

**Comment by vladikoff** [comment]
Manually created update for train 43 for now: https://github.com/mozilla/fxa-content-server/pull/2878

**Comment by vladikoff** [comment]
Going to close this for now and will bump manually

**Pull request closed without merging**